napari/layers/points/points.py

```diff
@@ -964,35 +873,93 @@ def face_color_mode(self):
 
     @face_color_mode.setter
     def face_color_mode(self, face_color_mode):
-        face_color_mode = ColorMode(face_color_mode)
+        self._set_color_mode(face_color_mode, 'face')
+
+    def _set_color_mode(
+        self, color_mode: Union[ColorMode, str], attribute: str
+    ):
+        """ Set the face_color_mode or edge_color_mode property
+
+        Parameters
+        ----------
+        color_mode : str, ColorMode
+            The value for setting edge or face_color_mode. If color_mode is a string,
+            it should be one of: 'direct', 'cycle', or 'colormap'
+        attribute : str
+            The name of the attribute to set the color of.
+            Should be 'edge' for edge_colo_moder or 'face' for face_color_mode.
+        """
+        color_mode = ColorMode(color_mode)
 
-        if face_color_mode == ColorMode.DIRECT:
-            self._face_color_mode = face_color_mode
-        elif face_color_mode in (ColorMode.CYCLE, ColorMode.COLORMAP):
-            if self._face_color_property == '':
+        if color_mode == ColorMode.DIRECT:
+            setattr(self, '_%s_color_mode' % attribute, color_mode)
+        elif color_mode in (ColorMode.CYCLE, ColorMode.COLORMAP):
+            color_property = getattr(self, '_%s_color_property' % attribute)
+            if color_property == '':
                 if self.properties:
-                    self._face_color_property = next(iter(self.properties))
+                    setattr(
+                        self,
+                        '_%s_color_property' % attribute,
```

Comment:
```suggestion
                        f'_{attribute}_color_property',
```

Also, since you use this multiple times in this function, it's probably worth creating it once, e.g. `color_property_attr = f'_{attribute}_color_property'` then using that each time. But maybe not worth it, as the string is more explicit. Either way, use f-strings. =)

Reply:
My preference is to write the string each time, because as you said, it's more explicit. However, I will change to f-strings.